src/interfaces/permissions-grant.ts

```diff
@@ -104,4 +106,34 @@ export class PermissionsGrant extends Message<PermissionsGrantMessage> {
       );
     }
   }
+
+  public static validateScope(permissionsGrantMessage: PermissionsGrantMessage): void {
+    const scope = permissionsGrantMessage.descriptor.scope;
+    if (scope.interface === DwnInterfaceName.Records) {
+      // RecordsRead and RecordsWrite
+      if (scope.method === DwnMethodName.Read || scope.method === DwnMethodName.Write) {
+        const recordsScope = scope as RecordsPermissionScope;
+
+        // `schema` scopes may not have protocol-related fields
+        if (recordsScope.schema !== undefined) {
+          if (recordsScope.protocol !== undefined || recordsScope.contextId !== undefined || recordsScope.protocolPath) {
+            throw new DwnError(
+              DwnErrorCode.PermissionsGrantScopeSchemaProhibitedFields,
+              'PermissionsGrants for RecordsRead and RecordsWrite that have `schema` present may not also have protocol-related properties present'
+            );
+          }
+        }
+
+        if (recordsScope.protocol !== undefined) {
```

Comment:
Sanity: does this imply `schema` is allowed when `protocol` is defined? Or this is disallowed else where in the code? (me lazy to hunt for it in code/tests)

Reply:
The previous `if` statement (`if (recordsScope.schema !== undefined)`), we make sure that `protocol` is undefined when `schema` is present.

I'll add tests to `PermissionsGrantHandler` to make sure that we check these conditions upon ingest.

Reply:
Added tests